### PR TITLE
Add option to authenticate using alternative GCE token source

### PR DESCRIPTION
**Pull request by 44past4**

### CODE REVIEW

File: prometheus-to-sd/kubernetes/prometheus-to-sd-kube-state-metrics.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ spec:
         memory: 300Mi
         cpu: 200m
   - name: prometheus-to-sd
-    image: gcr.io/google-containers/prometheus-to-sd:v0.9.0
+    image: gcr.io/google-containers/prometheus-to-sd:v0.9.1
```
[the remark this answers is not in the document]
Comment (44past4) [reply prometheus-to-sd/kubernetes/prometheus-to-sd-kube-state-metrics.yaml]: Fixed

**Comment by davidxia** [comment]
@x13n @44past4 how are the Docker images for prometheus-to-sd uploaded to Google Container Registry? This PR's commit \[1\] is marked 0.9.1, but I don't see 0.9.1 image here \[2\].

1: https://github.com/GoogleCloudPlatform/k8s-stackdriver/commit/2f7d151e4aaf2425fd69d1fe83472d7c6732f59c#diff-432c861ad3a967b20d12a8761c1c0d05
2: https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/prometheus-to-sd?gcrImageListsize=30

**Comment by x13n** [comment]
There is 0.9.1 tag on the commit, but the last release of prometheus-to-sd was 0.9.0. We need to cut a new release to get this PR included.

**Comment by davidxia** [comment]
That would be super helpful. Thanks!

**Comment by 44past4** [comment]
prometheus-to-sd 0.9.1 image has been published to the registry.

**Pull request merged**